src/utils/generateHypercertIdFromReceipt.ts

```diff
@@ -0,0 +1,47 @@
+import { decodeEventLog, getAddress, type TransactionReceipt } from "viem";
+import { HypercertMinterAbi } from "@hypercerts-org/contracts";
+
+export const generateHypercertIdFromReceipt = (
+  receipt: TransactionReceipt,
+  chainId: number,
+) => {
+  if (!receipt || !receipt.logs) {
+    console.log("No receipt found");
+    return;
+  }
+  const events = receipt.logs.map((log) =>
+    decodeEventLog({
+      abi: HypercertMinterAbi,
+      data: log.data,
+      topics: log.topics,
+    }),
+  );
+
+  if (!events) {
+    throw new Error("No events in receipt");
+  }
+
+  const claimEvent = events.find((e) => e.eventName === "ClaimStored");
+
+  if (!claimEvent) {
+    throw new Error("ClaimStored event not found");
+  }
+
+  const { args } = claimEvent;
+
+  if (!args) {
+    throw new Error("No args in event");
+  }
+
+  // @ts-expect-error it really does exist!
+  const tokenIdBigNumber = args["claimID"] as BigNumber;
```

Comment:
Why bignumber? We use js native bigint everywhere else

Reply:
Resolved in https://github.com/hypercerts-org/hypercerts-api/pull/177/commits/194cb7330f09d8396b50450b7bfcb8734b104212